Updates empty filter state messaging

AllanOXDi opened the pull request:
## Summary
This PR updates the logic and message displayed when the user list is empty due to active filters or search terms.

Previously, we were  using  the generic "No users exist" message.
closes [#13758](https://github.com/learningequality/kolibri/issues/13758)
## References
[#13758](https://github.com/learningequality/kolibri/issues/13758)

## Reviewer guidance
Navigate to > users  try searching for users that are not in the facility > filter roles that don't exist yet still.

nucleogenesis commented on packages/kolibri-common/strings/bulkUserManagementStrings.js:
```diff
@@ -461,6 +441,26 @@ export const bulkUserManagementStrings = createTranslator('BulkUserManagementStr
+  },
+
+  noUsersMatchFilter: {
+    message: 'No users match this filter',
```
When @marcellamaki noted:

> "No users match this filter" || "No users match these filters"

I think she meant that there would be differentiation between singular "this filter" and plural "these filters" when the user has selected only 1 vs more than 1 filter kinda like `'No users match {filtersCount, plural, one {this filter}, other {these filters}'`

nucleogenesis commented on packages/kolibri-common/strings/bulkUserManagementStrings.js:
```diff
@@ -461,6 +441,26 @@ export const bulkUserManagementStrings = createTranslator('BulkUserManagementStr
+  },
+
+  noUsersMatchFiltersAndSearch: {
+    message: 'No users match this search and this filter',
```
The other complication would be making this have a similar `filtersCount` property which changes between `this filter` and `these filters` depending on the value you pass into the string function.

nucleogenesis commented on packages/kolibri-common/strings/bulkUserManagementStrings.js:
```diff
@@ -461,6 +441,26 @@ export const bulkUserManagementStrings = createTranslator('BulkUserManagementStr
     context: 'Side panel H1 heading showing the number of selected users to assign as coaches',
   },
+
+  noUsersMatch: {
```
I don't see where this string is used.

nucleogenesis commented on kolibri/plugins/facility/assets/src/views/users/common/UsersTable.vue:
```diff
@@ -562,23 +560,14 @@
+            if (roleFilter.value && roleFilter.value.value !== ALL_FILTER) {
+              return noUsersMatchFiltersAndSearch$();
+            }
+            return noUsersMatchSearch$({ filterText: searchTerm.value });
```
Related to comments I left in the strings file - I think the logic here is right so you'd just have to pass in the proper number value of filters that are set ie `noUsersMatchFiltersAndSearch$({filtersCount: ...})` for this and the other message

nucleogenesis reviewed (dismissed):
I added some change requests to make the strings handle plural values for clearer language. The logic for showing the strings works as expected in my local testing.

pcenov commented:
@AllanOXDi @nucleogenesis - something is not right here, the users table is not loading at all now, and I am getting multiple errors in the console:

<img width="1360" height="852" alt="errors" src="https://github.com/user-attachments/assets/1f4ba590-0130-4e59-bf72-9e5036409738" />

AllanOXDi commented:
Let me fix that ASAP. I was a result of rebasing

pcenov commented:
@AllanOXDi - please let me know when this one is ready for testing again. At the moment I can see that there's no new build in the github-actions. Thanks!

nucleogenesis committed:
handcrafted rebasing of @AllanOXDi 's `filtersupdate` branch

Some significant changes were made in a PR that made some conflicts that
were difficult to resolve by picking chunks. I've opted instead to apply
the changes from Allan's PR directly onto `develop` after some failed
attempts at a regular rebase. This should provide the filters updates
expected from his original branch.

Co-authored-by: Allan Otodi Opeto
<103313919+AllanOXDi@users.noreply.github.com>

AllanOXDi commented:
@pcenov  This PR is now ready to be QA'ed again

AllanOXDi committed:
fixes failed  test

pcenov reviewed (approved):
Thanks @AllanOXDi, LGMT - implemented as specified in https://github.com/learningequality/kolibri/issues/13758.
